File: java/src/main/java/com/a65apps/kalimullinilnazrafilovich/interactors/notification/NotificationInteractor.java
package com.a65apps.kalimullinilnazrafilovich.interactors.notification;

import com.a65apps.kalimullinilnazrafilovich.entities.BirthdayNotification;
import com.a65apps.kalimullinilnazrafilovich.entities.ContactDetailsInfo;

public interface NotificationInteractor {
    BirthdayNotification onBirthdayNotification(ContactDetailsInfo contactDetailsInfo);

    BirthdayNotification offBirthdayNotification(ContactDetailsInfo contactDetailsInfo);

    BirthdayNotification getNotificationWorkStatus(ContactDetailsInfo contactDetailsInfo);

}

File: java/src/main/java/com/a65apps/kalimullinilnazrafilovich/interactors/notification/NotificationInteractor.kt
package com.a65apps.kalimullinilnazrafilovich.interactors.notification

import com.a65apps.kalimullinilnazrafilovich.entities.BirthdayNotification
import com.a65apps.kalimullinilnazrafilovich.entities.ContactDetailsInfo

interface NotificationInteractor {
    fun onBirthdayNotification(contactDetailsInfo: ContactDetailsInfo): BirthdayNotification

    fun offBirthdayNotification(contactDetailsInfo: ContactDetailsInfo): BirthdayNotification

    fun getNotificationWorkStatus(contactDetailsInfo: ContactDetailsInfo): BirthdayNotification
}
